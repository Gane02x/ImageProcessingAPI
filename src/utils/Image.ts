import sharp from "sharp"

export class Image {
  async edit(outdir: string, filename: string, width: string, height: string) {
    try {
      const newImage = await sharp(`./images/${filename}.jpg`)
        .resize(Number(width), Number(height))
        .toFile(outdir)
      return newImage
    } catch (error) {
      // eslint-disable-next-line no-console, no-undef
      console.log("Error at editing img", error)
    }
  }
}

export default new Image()
