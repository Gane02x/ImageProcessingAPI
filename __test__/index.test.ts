/* eslint-disable no-undef */

import { Server } from "http"
import request from "supertest"
import app from "../src/index"
import Image from "../src/utils/Image"

describe("Checking if the API handles each endpoint correctly", () => {
  const server: Server = app.listen()
  test("ImageCtrl handles the Resizing of images via URL", async () => {
    const res = await request(server).get(
      "/api/images?filename=GT500&width=1200&height=720"
    )
    expect(res.status).toEqual(200)
  })

  test("ImageCtrl displays an error if query if missing", async () => {
    const res = await request(server).get(
      "/api/images?filename=GT500?&width=1200&height"
    )
    expect(res.status).toEqual(400)
  })
  server.close()
})

describe("Image method .edit resizes img", () => {
  test(".edit resizes the Image with Sharp and returns the resized Image", async () => {
    const imgName = "M8"
    const width = "1200"
    const height = "720"
    const outdir = `./images/new/${imgName}-${width}-${height}.jpg`

    const res = await Image.edit(outdir, imgName, width, height)
    expect(res).toMatchSnapshot()
  })
})
