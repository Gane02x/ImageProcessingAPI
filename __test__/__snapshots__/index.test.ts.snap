// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image method .edit resizes img .edit resizes the Image with Sharp and returns the resized Image 1`] = `
Object {
  "channels": 3,
  "format": "jpeg",
  "height": 720,
  "premultiplied": false,
  "size": 104038,
  "width": 1200,
}
`;
